components/server/src/prebuilds/prebuild-manager.ts

```diff
@@ -365,10 +368,15 @@ export class PrebuildManager {
         return this.config.incrementalPrebuilds.repositoryPasslist.some((url) => trimRepoUrl(url) === repoUrl);
     }
 
-    async fetchConfig(ctx: TraceContext, user: User, context: CommitContext): Promise<WorkspaceConfig> {
+    async fetchConfig(
+        ctx: TraceContext,
+        user: User,
+        context: CommitContext,
+        configContext: WorkspaceConfigContext,
```

Comment:
We pass in the organizationId in many functions as first level context in addition to the user. Let's directly pass the `organizationId` as the third argument (i.e. after `user` instead of wrapping it in another shape.

Reply:
User already has organizationId, but it's the current org selected by user, not the workspace context, i.e. User can start workspaces from different org without change their current organization

Reply:
The user's organizationId is a static "belongsTo" relationship and is not meant to be used to do any actions on an org. It is also not set for users that are created and belong to installation level.

Reply:
@svenefftinge  Addressed https://github.com/gitpod-io/gitpod/pull/18723/commits/2756cafbad251e5522572d2a8e0f1b3e433fdef5